gson/src/main/java/com/google/gson/Gson.java

```diff
@@ -105,10 +105,14 @@
  * <p>See the <a href="https://github.com/google/gson/blob/main/UserGuide.md">Gson User Guide</a>
  * for a more complete set of examples.</p>
  *
- * <h2 id="default-lenient">Lenient JSON handling</h2>
+ * <h2 id="default-lenient">JSON Strictness handling</h2>
  * For legacy reasons most of the {@code Gson} methods allow JSON data which does not
- * comply with the JSON specification, regardless of whether {@link GsonBuilder#setLenient()}
- * is used or not. If this behavior is not desired, the following workarounds can be used:
+ * comply with the JSON specification when no explicit {@linkplain Strictness strictness} is set (the default).
+ * To specify the strictness of a {@code Gson} instance, you should set it through
+ * {@link GsonBuilder#setStrictness(Strictness)}.
+ *
+ * <p>For older Gson versions, which don't have the strictness mode API, the following
+ * workarounds can be used:
  *
  * <h3>Serialization</h3>
  * <ol>
```

Comment:
Maybe at the end of the `<h3>Deserialization</h3>` section below (after line 137), it should mention that this workaround is not fully strict, e.g.
```
 * <p>Note that the {@code JsonReader} created this way is only 'legacy strict', it mostly adheres
 * to the JSON specification but allows small deviations. See {@link JsonReader#setStrictness(Strictness)}
 * for details.
```

(unfortunately the GitHub UI does not let me comment there)

What do you think?